src/codeGenerator/store.ts

```diff
@@ -0,0 +1,103 @@
+const STORE: Map<string, FunctionValue | AdvancementValue> = new Map();
+export default STORE;
+
+export class FunctionValue {
+  name: string;
+  commands: string[];
+  tag: string | undefined;
+
+  constructor(name: string, commands: string[], tag: string | undefined) {
+    this.name = name;
+    this.commands = commands;
+    this.tag = tag;
+  }
+
+  static regular(name: string, commands: string[]) {
+    return new FunctionValue(name, commands, undefined);
+  }
+
+  static onTick(name: string, commands: string[]) {
+    return new FunctionValue(name, commands, "tick");
+  }
+
+  static onLoad(name: string, commands: string[]) {
+    return new FunctionValue(name, commands, "load");
+  }
+}
+
+export class AdvancementValue {
+  name: string;
+  title: string | undefined;
+  icon: string | undefined;
+  description: string | undefined;
+  parent: string | undefined;
+  rewardFunction: string | undefined;
+  triggers: Trigger[];
+
+  constructor(
+    name: string,
+    title: string | undefined,
+    icon: string | undefined,
+    description: string | undefined,
+    parent: string | undefined,
+    rewardFunction: string | undefined,
+    triggers: Trigger[],
+  ) {
+    this.name = name;
+    this.title = title;
+    this.icon = icon;
+    this.description = description;
+    this.parent = parent;
+    this.rewardFunction = rewardFunction;
+    this.triggers = triggers;
+  }
+}
+
+export abstract class Trigger {}
+
+export class ConsumeItem extends Trigger {
+  item: ItemSpec;
+
+  constructor(item: ItemSpec) {
+    super();
+    this.item = item;
+  }
+}
+
+export class InventoryChanged extends Trigger {
+  item: ItemSpec;
+
+  constructor(item: ItemSpec) {
+    super();
+    this.item = item;
+  }
+}
+
+export class Raw extends Trigger {
```

Comment:
```suggestion
export class RawTrigger extends Trigger {
```

Maybe this is a more specific name?

Reply:
No, on the grounds that every other trigger doesn't mention its parent class

Reply:
I would rather change them all to include Trigger tbh - class names should be clear and self contained, I'm convinced a module name of "Store" alone makes it clear that all of these are triggers.